src/cephadm/cephadm

```diff
@@ -3222,22 +3222,32 @@ def _pull_image(ctx, image):
 ##################################
 
 
-@infer_image
 def command_inspect_image(ctx):
     # type: (CephadmContext) -> int
+    try:
+        info_from = _inspect_image(ctx)
+    except RuntimeError as e:
+        print(str(e))
+        return errno.ENOENT
+
+    print(json.dumps(info_from, indent=4, sort_keys=True))
+    return 0
+
+
+@infer_image
+def _inspect_image(ctx):
+    # type: (CephadmContext) -> Dict[str, Union[str,List[str]]]
     out, err, ret = call_throws(ctx, [
         ctx.container_path, 'inspect',
         '--format', '{{.ID}},{{.RepoDigests}}',
         ctx.image])
     if ret:
-        return errno.ENOENT
+        raise RuntimeError('inspect image command failed: %s' % err)
```

Comment:
this can never happen:

https://github.com/ceph/ceph/blob/502409c96607b35004be3f18a694f2f33b8f46a9/src/cephadm/cephadm#L1455-L1465

Remove the `ret` value altogether?